src/Test/Mockery/Directory.hs

```diff
@@ -7,12 +7,19 @@ import           Control.Exception
 import           System.Directory
 import           System.IO.Temp (withSystemTempDirectory)
 
+-- |
+-- Run given action with the current working directory set to a temporary
+-- directory.  The temporary directory in created before the action is run and
```

Comment:
Maybe we should add a comment about the fact that the working directory is restored?